### PR TITLE
DOC: Docstring updated for DataFrame.equals

### LINKED ISSUE #34498: DOC: improve the .equals() docstring
The current docstring for Series/DataFrame `equals` method can use some improvement (https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.equals.html):

- " The column headers do not need to have the same type" is not very clear, and also doesn't apply to Series.equals. Something like "The row/column index do not need to have the same type (as long as the values are still considered equal)" might be better
- The "Notes" section is mostly duplicating the long description at the top.

### PR DESCRIPTION
**Pull request by pandeydeepak0**
- [ ] closes #34498

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -1247,9 +1247,10 @@ def equals(self, other):
 
         This function allows two Series or DataFrames to be compared against
         each other to see if they have the same shape and elements. NaNs in
-        the same location are considered equal. The column headers do not
-        need to have the same type, but the elements within the columns must
-        be the same dtype.
+        the same location are considered equal. The row/column index do not need
+        to have the same type (as long as the values are still considered equal).
+
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py:1252]: you can leave the blank line here

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR!

Can you also update or remove the "Notes" section?

File: pandas/core/generic.py [hunk 2]
```diff
@@ -1247,9 +1247,12 @@ def equals(self, other):
 
         This function allows two Series or DataFrames to be compared against
         each other to see if they have the same shape and elements. NaNs in
-        the same location are considered equal. The column headers do not
-        need to have the same type, but the elements within the columns must
-        be the same dtype.
+        the same location are considered equal.
+        
+        The row/column index do not need to have the same type, as long
+        as the values are still considered equal.
```
Comment (fujiaxiang) [inline pandas/core/generic.py:1253]: I feel the "still" can be removed. This was used in a different context under "Notes". It makes the sentence slightly confusing when used here. So we will have "as long as the values are considered equal".

**Review by fujiaxiang — CHANGES_REQUESTED** [review]

**Comment by erfannariman** [comment]
@freakypandit do you have time to finish this up?

File: pandas/core/generic.py (hunk 2 above)
Comment (jorisvandenbossche) [inline pandas/core/generic.py:1253]:
```suggestion
        as the values are considered equal.
```

Comment (jorisvandenbossche) [reply pandas/core/generic.py]: @fujiaxiang added your suggestion, thanks!

**Comment by simonjayhawkins** [comment]
@freakypandit can you fix the lint failure.

**Comment by freakypandit** [comment]
@simonjayhawkins I will try to do this.

**Comment by jorisvandenbossche** [comment]
I fixed the trailing whitespace (that was the linting failure)

**Review by simonjayhawkins — APPROVED** [review]
lgtm pending green

**Comment by jorisvandenbossche** [comment]
Thanks @freakypandit !

**Pull request merged**